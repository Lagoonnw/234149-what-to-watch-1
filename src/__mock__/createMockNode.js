export function createNodeMock(element) {
  if (element.type === `video`) {
    return {};
  }
  return null;
}
