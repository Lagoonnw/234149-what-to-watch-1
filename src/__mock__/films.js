export const films = [
  {
    id: 54585444,
    name: `Shutter Island`,
    genre: `thriller`,
    poster: `img/shutter-island.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 4587653,
    name: `Pulp Fiction`,
    genre: `crime`,
    poster: `img/pulp-fiction.jpg`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 54546847,
    name: `Snatch`,
    genre: `crime`,
    poster: `img/snatch.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 54545454545,
    name: `Bohemian Rhapsody`,
    genre: `drama`,
    poster: `img/bohemian-rhapsody.jpg`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 549963331,
    name: `Aviator`,
    genre: `family`,
    poster: `img/aviator.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

