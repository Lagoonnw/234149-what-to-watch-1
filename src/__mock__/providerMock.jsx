import React from 'react';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {genresList} from '../constants/constants';
import PropTypes from 'prop-types';

const mock = [
  {
    id: 54545454,
    name: `Snatch`,
    genre: `horror`,
    poster: `pic.jpg`,
    src: `video.mp4`,
    onMouseLeave: jest.fn(),
    onMouseEnter: jest.fn()
  },
  {
    id: 9898988,
    name: `Pulp Fiction`,
    genre: `horror`,
    poster: `pic.jpg`,
    src: `video.mp4`,
    onMouseLeave: jest.fn(),
    onMouseEnter: jest.fn()
  },
  {
    id: 545468787,
    name: `The Witcher`,
    genre: `horror`,
    poster: `pic.jpg`,
    src: `video.mp4`,
    onMouseLeave: jest.fn(),
    onMouseEnter: jest.fn()
  }
];
const initialStateMock = {
  activeGenre: null,
  movies: mock,
  sortedMovies: mock,
  genres: Array.from(genresList)
};
const reducer = (state = initialStateMock) => state;
const store = createStore(reducer);

export const ProviderMock = ({children}) => <Provider store={store}>{children}</Provider>;

ProviderMock.propTypes = {
  children: PropTypes.object
};
