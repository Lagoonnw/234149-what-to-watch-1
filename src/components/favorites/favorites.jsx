import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import Header from '../header/header.jsx';
import {connect} from 'react-redux';
import {getFavorites} from '../../reducers/user/selectors';
import {userAction} from '../../actions/user/action';
import {MoviesList} from '../movies-list/movies-list.jsx';
import {withActiveItem} from '../../hocs/with-active-item/with-active-item.jsx';

const MoviesListWithActiveItem = withActiveItem(MoviesList);

export class Favorites extends PureComponent {
  constructor(props) {
    super(props);
    props.fetchMovies();
  }

  render() {
    const {movies} = this.props;
    return (
      <div className="user-page">
        <Header className="user-page__head"/>
        <section className="catalog" style={{marginLeft: `16px`, marginRight: `16px`}}>
          <h2 className="catalog__title visually-hidden">Catalog</h2>
          <MoviesListWithActiveItem movies={movies}/>
        </section>
      </div>
    );
  }
}

Favorites.propTypes = {
  movies: PropTypes.array.isRequired,
  fetchMovies: PropTypes.func.isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  movies: getFavorites(state)
});

const mapDispatchToProps = (dispatch) => ({
  fetchMovies: () => dispatch(userAction.loadFavoriteMovies())
});

export default connect(mapStateToProps, mapDispatchToProps)(Favorites);
