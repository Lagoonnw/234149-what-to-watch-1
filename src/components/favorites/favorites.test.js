import React from 'react';
import renderer from 'react-test-renderer';
import {createNodeMock} from '../../__mock__/createMockNode';
import {ProviderMock} from '../../__mock__/providerMock';
import {Favorites} from './favorites';

const mockFilms = [
  {
    id: 54545454,
    name: `Snatch`,
    genre: `horror`,
    previewImage: `pic.jpg`,
    previewVideoLink: `video.mp4`,
  },
  {
    id: 9898988,
    name: `Pulp Fiction`,
    genre: `horror`,
    previewImage: `pic.jpg`,
    previewVideoLink: `video.mp4`,
  },
  {
    id: 545468787,
    name: `The Witcher`,
    genre: `horror`,
    previewImage: `pic.jpg`,
    previewVideoLink: `video.mp4`,
  }
];

describe(`Favorites component snapshot test`, () => {
  test(`Should render Favorites correctly`, () => {
    const options = {createNodeMock};
    const tree = renderer.create(
        <ProviderMock>
          <Favorites movies={mockFilms} fetchMovies={jest.fn()}/>
        </ProviderMock>
        , options);
    expect(tree).toMatchSnapshot();
  });
});
