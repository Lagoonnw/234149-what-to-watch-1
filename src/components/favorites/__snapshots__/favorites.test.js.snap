// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component snapshot test Should render Favorites correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/film/54545454"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                autoPlay={false}
                controls={false}
                height="175px"
                muted={true}
                poster="pic.jpg"
                src="video.mp4"
                style={
                  Object {
                    "objectFit": "cover",
                  }
                }
                width="280px"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/54545454"
              onClick={[Function]}
            >
              Snatch
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/film/9898988"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                autoPlay={false}
                controls={false}
                height="175px"
                muted={true}
                poster="pic.jpg"
                src="video.mp4"
                style={
                  Object {
                    "objectFit": "cover",
                  }
                }
                width="280px"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/9898988"
              onClick={[Function]}
            >
              Pulp Fiction
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/film/545468787"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                autoPlay={false}
                controls={false}
                height="175px"
                muted={true}
                poster="pic.jpg"
                src="video.mp4"
                style={
                  Object {
                    "objectFit": "cover",
                  }
                }
                width="280px"
              />
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/545468787"
              onClick={[Function]}
            >
              The Witcher
            </a>
          </h3>
        </article>
      </div>
    </div>
  </section>
</div>
`;
